### ISSUE TITLE
WebDAV support

### ISSUE THREAD

**Issue by KSXGitHub** [svenstaro/miniserve#293]
## Motivation

Sometimes I want to browse files hosted by miniserve the same way I do with local files (i.e. using a File Explorer).

## Suggestion

Implement [WebDAV](http://www.webdav.org/). Initially, read-only would be sufficed.

## Additional Suggestion

miniserve has a feature that allow end-user to upload their files. I think it should conform WebDAV standard.

**Comment by svenstaro** [comment]
Yeah sounds cool. I'll check out the rust WebDAV implementation. Maybe there's something we can already use.

**Comment by HerHde** [comment]
:wave: Hey, any updates on this one?

**Comment by Jikstra** [comment]
I just stumbled over and it has a actix 4 support. I tried out the example (https://github.com/miquels/webdav-handler-rs/blob/master/src/actix.rs) and upload/download/listing... seems to work. Quite suprised how well it works. I think we could use this as a middleware/service in miniserve.

**Comment by Dialga** [comment]
If anyone uses rclone, it has a builtin webdav server. It should be sufficient for a one off job.
`rclone serve webdav .`

**Comment by stokito** [comment]
a small webdav share program in plain C but also with Qt GUI  https://ltworf.github.io/weborf/

**Comment by svenstaro** [comment]
This is implemented as of #1415.

**Comment by ssokolow** [comment]
@stokito You might want to mention in the "File managers and tools" section that KDE's default file manager, Dolphin, has WebDAV support, as does Windows Explorer since the [Windows Desktop Update](https://en.wikipedia.org/wiki/Windows_Desktop_Update) for Windows 95 and NT 4.

**Comment by ssokolow** [comment]
\*nod* Also, as of Mac OS... 10.8, I think? It also has a WebDAV client built into Finder.

**Windows 98 SE:**
![Image](https://github.com/user-attachments/assets/658e6dab-eb2d-4328-b53d-5ef256b4f539)

**Windows XP:**
![Image](https://github.com/user-attachments/assets/4eee8b1b-3bb1-48d8-9dcf-f8ff2ad955b2)

**Mac OS 10.13 (High Sierra):**
![Image](https://github.com/user-attachments/assets/44ced57f-1578-4cca-b0fb-c9783785458e)

**Comment by ssokolow** [comment]
OK, I've confirmed that Mac OS X has WebDAV support built into Finder at least as far back as 10.1. I don't have a mac which can run 10.0 or the Rhapsody developer preview, but [this page](https://sfu.teamdynamix.com/TDClient/255/ITServices/KB/ArticleDet?ID=3937) says "Versions prior to 10.4" so 10.0 probably does too.

Given that [Apple themselves said](https://www.cnet.com/tech/computing/apple-finalizes-migration-to-webdav-for-idisks/) that you need to download a third-party WebDAV client to use post-AFP iDisks with Mac OS 9, I think it's safe to say it doesn't come with built-in support.
